plugins/providers/docker/action/create.rb

```diff
@@ -156,6 +155,16 @@ def forwarded_ports(include_ssh=false)
 
           result
         end
+
+        def container_name
+          base = @env[:root_path].basename.to_s
+          base = base.sub(/^[^a-zA-Z0-9]/, VAGRANTCONTAINER)
```

Comment:
Considering that the timestamp is appended to the container name below, collisions won't be a concern here. This means that instead of replacing an invalid first character, it can just be removed (the same as the invalid characters being removed from the full container name in the `gsub!` below).